identity_credential/src/validator/errors.rs

```diff
@@ -139,4 +140,65 @@ impl Display for CompoundPresentationValidationError {
   }
 }
 
-impl std::error::Error for CompoundPresentationValidationError {}
+impl Error for CompoundPresentationValidationError {}
+
+#[derive(Debug, thiserror::Error)]
+/// An error caused by a failure to verify a Domain Linkage configuration or credential.
+pub struct DomainLinkageVerificationError {
+  /// Cause of the error.
+  pub cause: DomainLinkageVerificationErrorCause,
+  /// Source of the error.
+  pub source: Option<Box<dyn Error + Send + Sync + 'static>>,
+}
+impl DomainLinkageVerificationError {
+  /// Provides the cause of the error.
+  pub fn get_cause(&self) -> &DomainLinkageVerificationErrorCause {
+    &self.cause
+  }
+}
+
+impl Display for DomainLinkageVerificationError {
+  fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+    write!(f, "{:?}; ", self.cause)?;
+    match &self.source {
+      Some(err) => write!(f, "{}", err)?,
+      None => {}
+    }
+    Ok(())
+  }
+}
+
+#[non_exhaustive]
+#[derive(Debug, thiserror::Error, strum::IntoStaticStr)]
+pub enum DomainLinkageVerificationErrorCause {
+  #[error("invalid credential")]
+  CredentialValidationError,
```

Comment:
This should probably wrap `CompoundCredentialValidationError` or something like that to give more information about what went wrong when validating the attached credential.

Reply:
From a logging perspective you are totally correct. 

The problem with the vague `CredentialValidation` variant is that one doesn't get an idea of which validations may fail by looking at the error. I am wondering whether it would make sense to either wrap `ValidationError` in the `CredentialValidation` variant (and call `CredentialValidator` with `FailFast::Yes` since domain linkage validation fails on the first error otherwise), or perhaps we include the relevant variants from  `ValidationError` as additional variants here. The latter is a bit more work to program, but would probably be nicer for TS users because more context becomes available with `Error.name`.  

Let me know what you prefer/think and also please free to chime in @PhilippGackstatter.